### DISCUSSION TITLE
Map panel how to use header stamp

### DISCUSSION THREAD

**Discussion by christian-nils** [foxglove/community#841] [Q&A]
Hello!

The Map panel is pretty easy to use but I wonder if there is a way to use the timestamp from the GPS header instead of the received timestamp? I have a significant offset there and it would be amazing if I could switch the type of timestamp used as there are for the Plot panel. Does this option exist? 

Thanks in advance

**Comment by jtbandes** [comment]
Could you clarify what you want to see when using the different timestamp? The points would still be in the same location on the map so I'm not sure what the visible difference would be.

**Reply by christian-nils** [reply]
Thanks for your question.  You can track the "current" location on the panel. But it uses the received time and i did not see any ways to change it.

**Reply by christian-nils** [reply]
An example:

[Screencast from 2023-10-13 08-42-23.webm](https://github.com/foxglove/community/assets/2658014/d5511faa-65f8-46fb-82f0-d3a66ee23e60)

I guess I can reprocess the file to change the received timestamp with the header timestamp, but I hoped there would be a configuration available to avoid this extra step :-)

**Reply by defunctzombie** [reply]
There is no such configuration option for the map panel. Could be worth adding such an option though I will note that "header stamp" is a ROS-ism. But there's probably a general way to have such a setting that works for ROS and non-ROS systems.
